### PR TITLE
fix: backup spend

### PR DESCRIPTION
**Pull request by SWvheerden**
Description
---
fixes pre mine backup spend

### CODE REVIEW

File: base_layer/wallet/src/output_manager_service/service.rs [hunk 1]
```diff
@@ -1663,14 +1670,14 @@ where
             range_proof_type,
             ..Default::default()
         };
-        let script = script!(PushPubKey(Box::new(recipient_address.public_spend_key().clone())));
+        let temp_script = script!(PushPubKey(Box::default()));
```
Comment (hansieodendaal) [inline base_layer/wallet/src/output_manager_service/service.rs:1668]: To be consistent
```suggestion
        let temp_script = push_pubkey_script(&PublicKey::default());
```

File: base_layer/wallet/src/output_manager_service/service.rs [hunk 2]
```diff
@@ -1691,7 +1698,7 @@ where
             .await?
             .with_sender_address(self.resources.one_sided_tari_address.clone())
             .with_recipient_data(
-                push_pubkey_script(recipient_address.public_spend_key()),
+                script!(PushPubKey(Box::default())),
```
Comment (hansieodendaal) [inline base_layer/wallet/src/output_manager_service/service.rs:1696]: ditto
```suggestion
                push_pubkey_script(&PublicKey::default()),
```

File: base_layer/wallet/src/output_manager_service/service.rs [hunk 3]
```diff
@@ -1791,7 +1804,7 @@ where
             .await?
             .with_input_data(ExecutionStack::default()) // Just a placeholder in the wallet
             .with_sender_offset_public_key(sender_offset_public_key)
-            .with_script_key(self.resources.key_manager.get_spend_key().await?.key_id)
+            .with_script_key(KeyId::Zero)
```
Comment (hansieodendaal) [inline base_layer/wallet/src/output_manager_service/service.rs:1802]:
```suggestion
            .with_script_key(KeyId::Zero)  // Just a placeholder in the wallet
```

File: base_layer/wallet/src/output_manager_service/service.rs [hunk 4]
```diff
@@ -1827,6 +1840,19 @@ where
...
+            return Err("Invalid identifier".to_string());
+        }
+        if data.len() % 2 != 0 {
+            return Err("Invalid payload".to_string());
+        }
+
+        let hex = identifier.to_owned() + data;
+        from_hex(hex.as_str()).map_err(|e| e.to_string())
+    }
+
```
Comment (hansieodendaal) [inline base_layer/wallet/src/output_manager_service/service.rs]: Is this used anywhere?

File: base_layer/wallet/src/output_manager_service/service.rs [hunk 5]
```diff
@@ -133,6 +135,8 @@ where
     TWalletConnectivity: WalletConnectivityInterface,
     TKeyManagerInterface: TransactionKeyManagerInterface,
 {
+    pub const PUSH_PUBKEY_IDENTIFIER: &str = "217e";
```
Comment (hansieodendaal) [inline base_layer/wallet/src/output_manager_service/service.rs]: Is this used anywhere?

**Review by hansieodendaal — COMMENTED** [review]
Some comments

File: base_layer/wallet/src/output_manager_service/service.rs (hunk 4 above)
Comment (SWvheerden) [reply base_layer/wallet/src/output_manager_service/service.rs]: good catch

**Pull request merged**